storey/targets.py

```diff
@@ -888,48 +922,87 @@ def _init(self):
             self._connection = taosws.connect(self._url)
         else:
             self._connection = taosws.connect(url=self._url, user=self._user, password=self._password)
+        self._closeables.append(self._connection)
         self._connection.execute(f"USE {self._database}")
 
+        self._number_of_values = len(self._columns) - self._number_of_tags
+        self._sql_template = self._get_sql_template()
+
     def _event_to_batch_entry(self, event):
         return self._event_to_writer_entry(event)
 
     @staticmethod
-    def _sanitize_value(value):
-        if isinstance(value, datetime.datetime):
-            value = round(value.timestamp() * 1000)
-        elif isinstance(value, str):
-            value = f"'{value}'"
-        return str(value)
+    def _get_params_template(num_param: int) -> str:
+        return f"({','.join(num_param * ['?'])})"
+
+    def _get_table_schema(
+        self, table_name: str
+    ) -> tuple[
+        list[tuple[str, Callable[[Any], "taosws.PyTagView"]]], list[tuple[str, Callable[[list], "taosws.PyColumnView"]]]
+    ]:
+        fields_data = [_TDEngineFieldData(*raw) for raw in self._connection.query(f"DESCRIBE {table_name};")]
+        tags_schema = []
+        reg_cols_schema = []
+        for field_data in fields_data:
```

Comment:
Maybe just call it
```suggestion
        for field in fields:
```
because `fields_data` isn't correct grammar, which makes it harder to understand.